### PR TITLE
Handle lost comms

### PR DESCRIPTION
**Pull request by BrianWoodruff**

### CODE REVIEW

**Review by avigupta33 — APPROVED** [review]

File: ovve_ui/utils/comms_link.py [hunk 1]
```diff
@@ -316,6 +317,13 @@ def run(self) -> None:
             self.logger.debug('Serial Init Successful')
         else:
             self.logger.error('Serial Initialization failed')
-            # When the alarm infrastructure is done this would trigger an alarm
+            # Signal the UI to display and sound an alarm
+            self.lost_comms_signal.emit()
```
Comment (mrrosen) [inline ovve_ui/utils/comms_link.py:321]: This actually might be a good use of the Hardware failure alarm on the UI-CU side; though maybe in the future

File: ovve_ui/utils/comms_link.py [hunk 2]
```diff
@@ -316,6 +317,13 @@ def run(self) -> None:
...
             return
-        self.process_SerialData()
+        
+        # Process serial data forever.  If we lose the connection,
+        # emit the the lost comms signal
+        try:
+            self.process_SerialData()
```
Comment (mrrosen) [inline ovve_ui/utils/comms_link.py:327]: `process_SerialData()` will not trigger an exception if it doesnt get anything for a while; so it doesnt really know the difference between receiving packets in a timely fashion and getting a single packet every 20 hours. Would probably need to modify this function to handle that. Will should at least trigger if the file handle for the serial port has a problem though, which is a comm error too.

**Review by mrrosen — COMMENTED** [review]
Just some thoughts, not just if what needs to change if anything right now

File: ovve_ui/utils/comms_link.py (hunk 2 above)
Comment (BrianWoodruff) [reply ovve_ui/utils/comms_link.py:327]: Ok, will check in this change for now and consider other ways to trigger the alarm.

**Pull request merged**